store/query_json.go

```diff
@@ -188,31 +193,26 @@ func (c *JSONCriterion) createcriterion(op JSONOperation, value interface{}) *JS
 
 // FindJSON queries for entities by JSONQuery
 func (t *Txn) FindJSON(q *JSONQuery) ([]string, error) {
-	dsq := dsquery.Query{
-		Prefix: baseKey.ChildString(t.model.name).String(),
-	}
-	dsr, err := t.model.store.datastore.Query(dsq)
+	txn, err := t.model.store.datastore.NewTransaction(true)
 	if err != nil {
-		return nil, fmt.Errorf("error when internal query: %v", err)
+		return nil, fmt.Errorf("error building internal query: %v", err)
 	}
+	defer txn.Discard()
+	iter := newIterator(txn, t.model.BaseKey(), q)
+	defer iter.Close()
 
 	var values []marshaledValue
 	for {
-		res, ok := dsr.NextSync()
+		res, ok := iter.NextSync()
 		if !ok {
 			break
 		}
+		// @fixme: This is the second time we unmarshal here, fix that!
```

Comment:
So we have two unmarshals per Query item result, since once is done on `newIterator`?

Reply:
At the moment, yes. But I think I'm going to push a quick change that actually just returns the unmarshalled object directly from the iterator instead to avoid this silly overhead.

Reply:
Fixed

Reply:
Nice!